Tests/TensorUtilsTests/WeightsTests.swift

```diff
@@ -0,0 +1,104 @@
+@testable import TensorUtils
+@testable import Hub
+import XCTest
+
+class WeightsTests: XCTestCase {
+
+    let downloadDestination: URL = {
+        FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!.appending(component: "huggingface-tests")
+    }()
+
+    var hubApi: HubApi { HubApi(downloadBase: downloadDestination) }
+
+    func testLoadWeightsFromFileURL() async throws {
+        let repo = "google/bert_uncased_L-2_H-128_A-2"
+        let modelDir = try await hubApi.snapshot(from: repo)
```

Comment:
I'd suggest you use [matching globs](https://github.com/huggingface/swift-transformers/blob/37e234e1ee4f2ef9c21c3d5cd592b5ec63d879af/Tests/HubTests/HubApiTests.swift#L188) to only download the two files we need and avoid additional downloads. There's a [version](https://github.com/huggingface/swift-transformers/blob/37e234e1ee4f2ef9c21c3d5cd592b5ec63d879af/Sources/Hub/HubApi.swift#L204) that supports multiple patterns.